api/main_endpoints/routes/Messages.js

```diff
@@ -0,0 +1,97 @@
+const {
+  UNAUTHORIZED,
+  FORBIDDEN,
+  SERVER_ERROR
+} = require('../../util/constants').STATUS_CODES;
+const express = require('express');
+const router = express.Router();
+const bodyParser = require('body-parser');
+const User = require('../models/User.js');
+
+router.use(bodyParser.json());
+
+const clients = {};
+
+const writeMessage = ((roomId, message) => {
+  if (clients[roomId]) {
+    clients[roomId].forEach(client => client.res.write(`data: ${JSON.stringify(message)}\n\n`));
+  }
+});
+
+router.post('/send', async (req, res) => {
+
+  const {apiKey, message, id} = req.body;
+
+  let apiKeyFound = false;
+
+  await User.findOne({apiKey}, (error, result) => {
+    if (error) {
+      res.sendStatus(SERVER_ERROR);
+      return;
+    }
+    if (result) {
+      apiKeyFound = true;
+    }
+  });
+
+  if(!apiKey || !message || !id){
+    res.sendStatus(FORBIDDEN);
+    return;
```

Comment:
this if statement should go before the user.findOne, leave the else if where it is

Reply:
ryan is solving the issue rn lets gooooo!